AMM/contracts/AMM/tests/functions/add_pool.rs

```diff
@@ -0,0 +1,89 @@
+use crate::utils::{
+    amm_abi_calls::{add_pool, pool},
+    test_helpers::{deploy_and_construct_exchange_contract, setup, setup_and_initialize},
+};
+
+mod success {
+    use super::*;
+
+    #[tokio::test]
+    async fn adds() {
+        let (wallet, amm_instance, assets) = setup_and_initialize().await;
+        let pair = (assets[0], assets[1]);
+        let exchange_contract_id =
+            deploy_and_construct_exchange_contract(&wallet, pair, None).await;
+        add_pool(&amm_instance, pair, exchange_contract_id).await;
```

Comment:
I would add a check before you add a pool that the `exchange_contract_id_in_storage == None`

Reply:
Actually nvm, you test this in another case.